### DISCUSSION TITLE
Saving temporary data on entities(specifically players)

### DISCUSSION THREAD

**Discussion by Braayy** [Minestom/Minestom#1100] [Help] [answered]
Is there a way of saving temporary data in entities, like the metadata api on Bukkit. I know there is the Tag api but it is more for persistent data i think

**Comment by GoldenCookieGIT** [comment]
tags are not persistent

**Reply by GoldenStack** [reply]
For anyone reading: Tags are basically a wrapper around NBT tags, and nothing in Minestom is intrinsically persistent (i.e. nothing is saved in files or anything), so tags are your best bet for this.

**Reply by MrlnHi** [reply]
While they are indeed a wrapper around NBT, they also cache their values so they don't have to initialize whatever they hold everytime you need it

**Reply by Braayy** [reply]
I know that tags aren't persistent by default because minestom by itself does not implement loading and saving, but if something(a custom implementation or extension) does this kind of work, it will save/load the entity's tags.

**Comment by DasLixou** [comment]
Just make a List of Tags as a Tag and call it tempdata. everytime before saving, just remove them and readd them after

**Comment by VegetalDev** [comment, marked as answer]
If you just want to associate arbitrary data with an entity, or the tag API is too limiting for your use case, you can also try WeakHashMap<Entity, [your data type]>. It'll automatically remove key-value entries as the Entity objects are garbage collected, and won't keep any strong references to them on its own.

**Reply by Braayy** [reply]
Yeah, i was doing something like this, but not using a weakhashmap(which is pretty adequate for my purposes, thanks!). I was asking just because i tought there was some other/better way
